printing/postprocess.py

```diff
@@ -1,21 +1,36 @@
 import os
 import re
 import subprocess
+import sys
 
-from control.models import PrintingProperties
-from printing import SANDBOX_PATH
+from control.models import PrintJob, JobStatus
+from printing import SANDBOX_PATH, JobCancelledException
 
 
-def postprocess_postscript(input_file: str, work_dir: str, properties: PrintingProperties):
+def _no_pages_cancel(job):
+    job.status = JobStatus.CANCELED
+    job.status_reason = 'No pages to print. Check your pages filter expression.'
+    job.save()
+    raise JobCancelledException()
+
+
+def postprocess_postscript(input_file: str, work_dir: str, job: PrintJob):
     out = os.path.join(work_dir, 'final.pdf')
-    subprocess.check_call([SANDBOX_PATH, work_dir, 'gs', '-sDEVICE=pdfwrite', '-dNOPAUSE',
-                           '-dBATCH', '-dSAFER', '-dCompatibilityLevel=1.4',
-                           '-sOutputFile=' + out, input_file])
+    subprocess.check_output([SANDBOX_PATH, work_dir, 'gs', '-sDEVICE=pdfwrite', '-dNOPAUSE',
+                             '-dBATCH', '-dSAFER', '-dCompatibilityLevel=1.4',
+                             '-sOutputFile=' + out, input_file], stderr=subprocess.STDOUT)
 
+    properties = job.properties
     if properties.pages_to_print:
         new_out = os.path.join(work_dir, 'post.pdf')
-        pages = properties.pages_to_print.replace(',', ' ').split()
-        subprocess.check_call([SANDBOX_PATH, work_dir, 'pdftk', out, 'cat', *pages, 'output', new_out])
+        pages = properties.pages_to_print.split(',')
```

Comment:
Is `properties.pages_to_print` somehow normalized here? Otherwise we'll handle arguments such as `1 2,4` incorrectly.

Reply:
I forgot to migrate server-side validation from the old form-based UI. Fixed, not it is required to be in a specific format.